src/classes/RD2_ScheduleService.cls

```diff
@@ -0,0 +1,120 @@
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Class with methods to create, modify and visualize RecurringDonationSchedule__c records.
+*
+*/
+
+public with sharing class RD2_ScheduleService {
+
+    private npe03__Recurring_Donation__c parentRecurringDonation;
+    @TestVisible
+    private List<RecurringDonationSchedule__c> recurringDonationSchedules = new List<RecurringDonationSchedule__c>();
+
+    private void init(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdSchedules) {
+        parentRecurringDonation = rd;
+        if (rdSchedules != null) {
+            recurringDonationSchedules = rdSchedules;
+        }
+    }
+
+    public RD2_ScheduleService(npe03__Recurring_Donation__c rd) {
+        init(rd, null);
+    }
+
+    public RD2_ScheduleService(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdSchedules) {
+        init(rd, rdSchedules);
+    }
+
+    /*******************************************************************************************************
+    * @description Initializes RecurringDonationSchedule__c record(s) associated with a parent
+    * npe03__Recurring_Donation__c record (parentRecurringDonation)
+    * @return List<RecurringDonationSchedule__c>
+    */
+    public List<RecurringDonationSchedule__c> initializeSchedule() {
+        if (parentRecurringDonation != null && parentRecurringDonation.Id != null && parentRecurringDonation.Status__c != 'Closed') {
+            recurringDonationSchedules.add(new RecurringDonationSchedule__c());
+            if (parentRecurringDonation.Day_of_Month__c == '1st and 15th' && parentRecurringDonation.npe03__Installment_Period__c == 'Monthly') {
+                recurringDonationSchedules.add(new RecurringDonationSchedule__c());
```

Comment:
I think I see how you're building this for 1st and 15th, but I think it may be limiting. At the moment, we only know of a single situation where there will be two schedule records. However, that could change at some point. Rather than add a blank sobject to the collection and then call a method that populates the fields in that blank sobject. I would eliminate the `recurringDonationSchedules` public var and just have the generateRecurringDonationScheduleDetails() build and return the correct number of records.

Reply:
All great suggestions.  Thanks!!!